packages/babel-parser/src/plugins/typescript/index.js

```diff
@@ -621,6 +621,11 @@ export default (superClass: Class<Parser>): Class<Parser> =>
 
       this.expect(tt.bracketL);
       node.typeParameter = this.tsParseMappedTypeParameter();
+
+      if (this.eatContextual("as")) {
+        node.nameType = this.tsParseType();
```

Comment:
If we don't see a `nameType`, `node.nameType` should be nullish otherwise it contradicts with the type definitions.